saleor/order/utils.py

```diff
@@ -355,18 +377,47 @@ def add_variant_to_order(
             is_shipping_required=variant.is_shipping_required(),
             quantity=quantity,
             unit_price=unit_price,
+            undiscounted_unit_price=undiscounted_unit_price,
             total_price=total_price,
+            undiscounted_total_price=undiscounted_total_price,
             variant=variant,
         )
-        unit_price = manager.calculate_order_line_unit(order, line, variant, product)
-        total_price = manager.calculate_order_line_total(order, line, variant, product)
-        line.unit_price = unit_price
-        line.total_price = total_price
-        line.undiscounted_unit_price = unit_price
-        line.undiscounted_total_price = total_price
+        unit_price_data = manager.calculate_order_line_unit(
+            order, line, variant, product
+        )
+        total_line_price_data = manager.calculate_order_line_total(
+            order, line, variant, product
+        )
+        line.unit_price = unit_price_data.price_with_discounts
+        line.total_price = total_line_price_data.price_with_discounts
+        line.undiscounted_unit_price = unit_price_data.undiscounted_price
+        line.undiscounted_total_price = total_line_price_data.undiscounted_price
         line.tax_rate = manager.get_order_line_tax_rate(
             order, product, variant, None, unit_price
         )
+
+        unit_discount = line.undiscounted_unit_price - line.unit_price
+        if unit_discount.gross:
+            sale_id = get_sale_id_applied_as_a_discount(
+                product=product,
+                price=channel_listing.price,
+                discounts=discounts,
+                collections=collections,
+                channel=channel,
+                variant_id=variant.id,
+            )
+            taxes_included_in_prices = site_settings.include_taxes_in_prices
+            if taxes_included_in_prices:
+                discount_amount = unit_discount.gross
+            else:
+                discount_amount = unit_discount.net
+            line.unit_discount = discount_amount
+            line.unit_discount_value = discount_amount.amount
+            line.unit_discount_reason = (
+                f"Sale: {graphene.Node.to_global_id('Sale', sale_id)}"
+            )
+            line.sale_id = sale_id
```

Comment:
To be consistent with the way how we handle it in conversion checkoutLine->orderLine
`sale_id=graphene.Node.to_global_id("Sale", sale_id) if sale_id else None,`